src/Cameras/Inputs/BaseCameraPointersInput.ts

```diff
@@ -324,7 +324,7 @@ export abstract class BaseCameraPointersInput implements ICameraInput<Camera> {
      * release.
      * Override this method to provide functionality.
      */
-    protected onButtonUp(evt: PointerEvent): void {
+    protected onButtonUp(evt: PointerEvent, buttonCount: number): void {
```

Comment:
why that change?

Reply:
2 reasons:
 1. Derived classes might well need to know how many buttons are down. Much easier to pass that up from the base (which already knows) than count presses and releases in the derived class.
 2. It's also a useful indicator for that multiple button press state has been cleared up.

Reply:
sure but this information is already in the PointerEvent no?

Reply:
hrm. Fair point.
So there is the subtle difference that `buttonCount` is only the number of buttons that `BaseCameraPointersInput` is aware of (maximum 2) whereas `evt: PointerEvent` presumably returns /all/ depressed buttons.
I think the user likely does not care about this subtlety though.

So, since this is only useful in testing when we care about how many buttons `BaseCameraPointersInput` is aware of ,
i think you are going to ask me to remove it....

Removing this will mean we have to either:
  1. not test how the Base class tracks button presses, or
  2. make some other attribute public (which IMO is the same as passing `buttonCount` here).

Personally i prefer to see things well tested so would like this to remain.

There is another instance of this in `onButtonDown(evt: PointerEvent, buttonCount: number)`.
If we remove it here, the same argument applies to both.

Reply:
Actually,
a compromise would be to keep a public member variable `buttonCount: number` but not document it.

Reply:
well not a big fan of that at all :) I really recommend removing both. the number of buttons to take care of should be something we may want in tests so let's handle it in tests where we can hack around easily without polluting the main code base

Reply:
up?

Reply:
?

Reply:
Did you remove both? I may have missed the commit  to be honest :)

Reply:
I would like to get all stuff related to tests in the test file itself to not pollute the main code

Reply:
> Did you remove both? I may have missed the commit to be honest :)

Not yet. Been busy with other things.
soon....

Reply:
right, done.
So i've made the variables that store the latest button press events (`pointA` and `pointB`) `private` members of `BaseCameraPointersInput`.

This way we can do horrible hacky things to access them in tests without them polluting the TS namespace.

Reply:
Well this does not solve my issue as we should not have ANYTHING that is not used in the core. I know that this is painful from your standpoint but it looks artificial (because it is :D) to have this properties but not use them

Reply:
just to confirm, these are existing variables. The Base class needs them.

I have made them slightly more accessible by moving them outside the `attachControl()` method though.

Reply:
And please, if that's still not acceptable, just say so.
I'll make fun of you but i won't mind.

Reply:
lol no worries! It is tough to get me upset :)

Reply:
Oh and yeah for the points! I'm fine with it! Ok to merge for me
Waiting for your final go